utils/Recorder/Recorder.ts

```diff
@@ -0,0 +1,95 @@
+import { ContractReceipt, UnsignedTransaction } from "ethers";
+import { HardhatRuntimeEnvironment } from "hardhat/types";
+import { getDeployer, getProposer, getTimelock } from "../DeployedContractsHelpers";
+import { Strategy } from "./strategies/Strategy.interface";
+import { TimelockStrategy } from "./strategies/TimelockStrategy";
+import { ImmediateExecutionStrategy } from "./strategies/ImmediateExecutionStrategy";
+import { SignerWithAddress } from "hardhat-deploy-ethers/dist/src/signers";
+import { blockTimeSeconds } from "../Constants";
+import { Signer } from "ethers";
+
+// class responsible for recoding and executing transactions
+// construction:
+// strategy - that will be used to execute transactions
+export class Recorder {
+  recordedTransactions: UnsignedTransaction[] = [];
+  executedTransactions: ContractReceipt[] = [];
+  strategy: Strategy;
+
+  constructor(_strategy: Strategy) {
+    this.strategy = _strategy;
+  }
+
+  // pushes an unsigned transaction (the parameter) to the recordedTransaction List
+  record(transaction: UnsignedTransaction) {
+    this.recordedTransactions.push(transaction);
+  }
+
+  // execute all recorderTransaction in order using strategy and clears recordedTransaction
+  async execute(): Promise<ContractReceipt[]> {
+    const transactionsToExecuteCount = this.recordedTransactions.length;
+
+    const response = await this.strategy.execute(this.recordedTransactions);
+    this.recordedTransactions = [];
+    this.executedTransactions.concat(response);
+
+    console.log(`${transactionsToExecuteCount} transactions were executed`);
+    return response;
+  }
+
+  print() {
+    console.log(this.recordedTransactions);
+  }
+}
+
+type DefaultRecorderParams = {
+  executionStartInDays: number | null;
+  singer: SignerWithAddress | null;
+};
+
+export async function defaultRecorder(hre: HardhatRuntimeEnvironment, params: DefaultRecorderParams | null = null) {
+  if (["mainnetFork", "arbitrumTestnet", "goerli", "kovan"].includes(hre.network.name)) {
```

Comment:
NIT
Maybe move that to the constants file right under the chain ids.. That way when adding a new chain we won't forget to also check this out.